Plugin/CheckoutLayoutPlugin.php

```diff
@@ -38,14 +61,35 @@ public function afterProcess(\Magento\Checkout\Block\Checkout\LayoutProcessor $p
                 'id' => 'kl_sms_consent',
             ];
 
-            $jsLayout['components']['checkout']['children']['steps']['children']['shipping-step']['children']['shippingAddress']['children']['shipping-address-fieldset']['children']['kl_sms_consent'] = $smsConsentCheckbox;
+            $address = $this->_getDefaultAddressIfSetForCustomer();
+
+            if (!$address)
+                $jsLayout['components']['checkout']['children']['steps']['children']['shipping-step']['children']['shippingAddress']['children']['shipping-address-fieldset']['children']['kl_sms_consent'] = $smsConsentCheckbox;
+            else {
+
+                // extra un-editable field with saved phone number to display to logged in users with default address set
+                $smsConsentTelephone = [
+                    'component' => 'Magento_Ui/js/form/element/abstract',
+                    'config' =>
+                        [
+                            'customScope' => 'shippingAddress',
+                            'template' => 'ui/form/field',
+                            'elementTmpl' => 'ui/form/element/input',
+                        ],
+                    'label' => 'Phone Number',
+                    'provider' => 'checkoutProvider',
+                    'sortOrder' => '120',
+                    'disabled' => true,
+                    'visible' => true,
+                    'value' => $address->getTelephone()
```

Comment:
Is it possible for a customer to have a saved address without a phone number?

Reply:
@njparadis no telephone is a required property when saving an address